website_sale_custom_filter/models/product.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2022 Camptocamp SA
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html)
+
+from odoo import api, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    @api.model
+    def _search_get_detail(self, website, order, options):
+        res = super()._search_get_detail(website, order, options)
+        # modify base_domain to include selected filters
```

Comment:
Would be good to explain a bit why you select the first element in `base_domain`.